8216507: StyleablePropertyFactory: example in class javadoc does not compile

karthikpandelu opened the pull request:
In the javadoc example of `StyleablePropertyFactory` class, wrong parameter was used while initializing `StyleableProperty`. 

Assigned the `CssMetadData` object to `SELECTED` variable returned by `createBooleanCssMetaData` method called in anonymous inner-class class, which is then used while initializing `StyleableProperty`.

Added missing semicolon in the same example.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8216507](https://bugs.openjdk.org/browse/JDK-8216507): StyleablePropertyFactory: example in class javadoc does not compile


### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - Committer)
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/983/head:pull/983` \
`$ git checkout pull/983`

Update a local copy of the PR: \
`$ git checkout pull/983` \
`$ git pull https://git.openjdk.org/jfx pull/983/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 983`

View PR using the GUI difftool: \
`$ git pr show -t 983`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/983.diff">https://git.openjdk.org/jfx/pull/983.diff</a>

</details>

andy-goryachev-oracle commented on modules/javafx.graphics/src/main/java/javafx/css/StyleablePropertyFactory.java:
```diff
@@ -121,12 +121,14 @@ public final class MyButton extends Button {
  * <pre><code>
  public final class MyButton extends Button {
 
+     private static CssMetaData<JavadocFixStylableProperty, Boolean> SELECTED;
```
should it be

```
private static CssMetaData<MyButton, Boolean> SELECTED;
```

in order to compile?

karthikpandelu replied:
Updated this change

andy-goryachev-oracle commented on modules/javafx.graphics/src/main/java/javafx/css/StyleablePropertyFactory.java:
```diff
@@ -141,7 +143,7 @@ public final class MyButton extends Button {
      // StyleableProperty implementation reduced to one line
      private final {@literal StyleableProperty<Boolean>} selected =
-         new SimpleStyleableBooleanProperty(this, "selected", "my-selected");
+         new SimpleStyleableBooleanProperty(SELECTED, "selected", "my-selected");
```
as an exercise, could you actually put MyButton into a scene graph and style it with CSS?

karthikpandelu replied:
I will check this and update here

andy-goryachev-oracle replied:
Do you think it's worth doing this as a unit test?

karthikpandelu replied:
I think it is good to add unit test to test the example code. Only thing I'm looking at is that if the example changes in future, we need to make sure that corresponding `MyButton` class is updated in unit test. 
Since the example code might not change frequently, we can add unit test.

andy-goryachev-oracle replied:
I agree, let's add a unit test, unless  `StyleablePropertyFactoryTest` covers the same scenario.

karthikpandelu replied:
Added unit test for the 3rd example in `StyleablePropertyFactoryTest`.
Unit test for 1st example is already covered as part of `myBoolean` property.
Since example 2 and 3 are similar except for the place where `createBooleanCssMetaData` method is called, added single test for this.
Please review the newly added unit test

andy-goryachev-oracle commented:
sorry, did not see this PR is still in a Draft stage...   why did I get a notification then?

also, jcheck is failing.

karthikpandelu commented:
It was in rfr stage. i just saw the build failure and converted it to draft before addressing the build error. 
Fixed the build issue now and moved to rfr again. Sorry for the confusion.

aghaisas reviewed (approved):
Looks good to me!

andy-goryachev-oracle reviewed (approved):
good work!

karthikpandelu commented:
/integrate

andy-goryachev-oracle commented:
/sponsor
